addons/purchase/models/res_partner.py

```diff
@@ -12,7 +12,6 @@ class res_partner(models.Model):
     def _compute_purchase_order_count(self):
         # retrieve all children partners and prefetch 'parent_id' on them
         all_partners = self.with_context(active_test=False).search([('id', 'child_of', self.ids)])
```

Comment:
If you're going to remove the prefetch, then you should update the code comment to match.

Also, this code is super old, I don't understand why it needed to prefetch the `parent_id` in the first place or why it wasn't just using a `read_group` to only get the `id` considering it doesn't use `all_partners` for anything else.

All that being said, it seems like this issue was already fixed in 17+ via this: https://github.com/odoo/odoo/pull/112126/commits/83b9f7401ab32da1cb8ea5a99e04ad9f4d7ce0fe#diff-399ff553a811afbc017518d8ca7992de980422f6e9683504b4d3fe5183e22377 so it's maybe not worth changing in v16 for this edge case (500K+ children is a bit much), especially if the customer can just upgrade to v17 to fix the issue